### PR TITLE
Fix collision counter in case some player names are reserved with cookies

### PR DESCRIPTION
**Pull request by o01eg**
`ServerApp::IsAvailableName` checks names in cookies as well so it possible loop doesn't find available name if counter use only player from lobby data. PR defines addition tries depending on size of stored cookie.

This PR fixes issue in v0.4.8 so should be cherry-picked there.

### CODE REVIEW

File: server/ServerFSM.cpp [hunk 1]
```diff
@@ -852,7 +852,9 @@ sc::result MPLobby::react(const JoinGame& msg) {
 
         bool collision = true;
         std::size_t t = 1;
-        while (t <= m_lobby_data->m_players.size() + 1 && collision) {
+        while (t <= m_lobby_data->m_players.size() + server.Networking().GetCookiesSize() + 1 &&
```
Comment (geoffthemedio) [inline server/ServerFSM.cpp]: reorder to
````
(collision &&
 t <= m_lobby_data ...
````

and indent to match

Comment (o01eg) [reply server/ServerFSM.cpp]: Fixed.

**Comment by Vezzra** [comment]
@geoffthemedio, @o01eg, anything preventing this from being merged?

**Comment by geoffthemedio** [comment]
no

**Pull request merged**